internal/host/hostcommands/free_addresses_cmd_test.go

```diff
@@ -17,48 +17,89 @@ var _ = Describe("free_addresses", func() {
 	var db *gorm.DB
-	var fCmd *freeAddressesCmd
+	var fCmd CommandGetter
 	var id, clusterId strfmt.UUID
 	var stepReply []*models.Step
 	var stepErr error
 	var dbName string
 
-	BeforeEach(func() {
-		db, dbName = common.PrepareTestDB()
-		fCmd = NewFreeAddressesCmd(common.GetTestLog(), "quay.io/ocpmetal/free_addresses:latest")
+	Context("Enabled", func() {
+		BeforeEach(func() {
+			db, dbName = common.PrepareTestDB()
+			fCmd = newFreeAddressesCmd(common.GetTestLog(), "quay.io/ocpmetal/free_addresses:latest", true)
 
-		id = strfmt.UUID(uuid.New().String())
-		clusterId = strfmt.UUID(uuid.New().String())
-		host = hostutil.GenerateTestHost(id, clusterId, models.HostStatusInsufficient)
-		host.Inventory = common.GenerateTestDefaultInventory()
-		Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
-	})
+			id = strfmt.UUID(uuid.New().String())
+			clusterId = strfmt.UUID(uuid.New().String())
+			host = hostutil.GenerateTestHost(id, clusterId, models.HostStatusInsufficient)
+			host.Inventory = common.GenerateTestDefaultInventory()
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+		})
 
-	It("happy flow", func() {
-		stepReply, stepErr = fCmd.GetSteps(ctx, &host)
-		Expect(stepReply).ToNot(BeNil())
-		Expect(stepReply[0].StepType).To(Equal(models.StepTypeFreeNetworkAddresses))
-		Expect(stepErr).ShouldNot(HaveOccurred())
-	})
+		It("happy flow", func() {
+			stepReply, stepErr = fCmd.GetSteps(ctx, &host)
+			Expect(stepReply).ToNot(BeNil())
+			Expect(stepReply[0].StepType).To(Equal(models.StepTypeFreeNetworkAddresses))
+			Expect(stepErr).ShouldNot(HaveOccurred())
+		})
 
-	It("Illegal inventory", func() {
-		host.Inventory = "blah"
-		stepReply, stepErr = fCmd.GetSteps(ctx, &host)
-		Expect(stepReply).To(BeNil())
-		Expect(stepErr).To(HaveOccurred())
-	})
+		It("Illegal inventory", func() {
+			host.Inventory = "blah"
+			stepReply, stepErr = fCmd.GetSteps(ctx, &host)
+			Expect(stepReply).To(BeNil())
+			Expect(stepErr).To(HaveOccurred())
+		})
+
+		It("Missing networks", func() {
+			host.Inventory = "{}"
+			stepReply, stepErr = fCmd.GetSteps(ctx, &host)
+			Expect(stepReply).To(BeNil())
+			Expect(stepErr).To(HaveOccurred())
+		})
 
-	It("Missing networks", func() {
-		host.Inventory = "{}"
-		stepReply, stepErr = fCmd.GetSteps(ctx, &host)
-		Expect(stepReply).To(BeNil())
-		Expect(stepErr).To(HaveOccurred())
+		AfterEach(func() {
+			// cleanup
+			common.DeleteTestDB(db, dbName)
+			stepReply = nil
+			stepErr = nil
+		})
 	})
+	Context("Disabled", func() {
+		BeforeEach(func() {
+			db, dbName = common.PrepareTestDB()
+			fCmd = newFreeAddressesCmd(common.GetTestLog(), "quay.io/ocpmetal/free_addresses:latest", false)
+
+			id = strfmt.UUID(uuid.New().String())
+			clusterId = strfmt.UUID(uuid.New().String())
+			host = hostutil.GenerateTestHost(id, clusterId, models.HostStatusInsufficient)
+			host.Inventory = common.GenerateTestDefaultInventory()
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
```

Comment:
Looks like too much of duplicate code, maybe you could reuse the same initialization except for the enabled/disabled part?

Reply:
The expect is also different